clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/AbstractStorageEngine.java

```diff
@@ -88,6 +89,18 @@ public AbstractStorageEngine(
     this.partitionStateSerializer = partitionStateSerializer;
   }
 
+  /**
+   * Making it public is for testing purpose.
+   */
+  public ReadWriteLock getRWLockForPartitionOrThrow(int partitionId) {
```

Comment:
For unit test testing purpose, it can be protected or default level should be fine.

Reply:
Actually, `StoreIngestionTaskTest` needs to mock this method, so it has to be `public` with current structure.